packages/edit-site/src/store/actions.js

```diff
@@ -147,6 +148,7 @@ export function* showHomepage() {
 
 	const page = {
 		path: '/',
+		label: showOnFront === 'page' ? __( 'Homepage' ) : __( 'All Posts' ),
```

Comment:
could just change this to homepage. (and then also update that in the page selector)